cpp/core/jni/JniCommon.h

```diff
@@ -166,27 +166,29 @@ class SparkAllocationListener final : public gluten::AllocationListener {
  public:
   SparkAllocationListener(
       JavaVM* vm,
-      jobject javaListener,
-      jmethodID javaReserveMethod,
-      jmethodID javaUnreserveMethod,
+      jobject jListenerLocalRef,
```

Comment:
We should limit usage, create listener from java call native, and pass local ref in, only convert to global ref in listener internal, let listener manage this ref, otherwise, we can not manage this ref easily.

I'm not sure whether newGlobalRef is idempotent...